account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -291,17 +301,40 @@ def _update_statement_balance_values(
         statement_date_since,
         statement_date_until
     ):
-        """Update statement start or end balance from line data."""
+        """Update statement start or end balance from line data.
+
+        Statements retrieved and statements created might have different begin
+        and end dates. Suppose we retrieve data that covers three days, and has four
+        transactions, but the statement to be created is for only one day. like this:
+        statement_values: {
+            "balance_start": 25.00,
+            "balance_end_real": 150.00,
+        }
+        lines_data = [
+            {"date": "2020-02-01", "amount": 25.00},
+            {"date": "2020-02-02", "amount": 25.00},
+            {"date": "2020-02-02", "amount": 50.00},
+            {"date": "2020-02-03", "amount": 25.00},
+        ]
+        If we create a statement for only the day 2020-02-02, the first and
+        the last line should not belong to this statement. But at the start of
+        2020-02-02 the balance was 50.00. So the amounts for lesser dates should
+        be added to the balance_start. Similarly amounts from later dates should
+        be subtracted from balance_end_real. Which should be 125.00 for
+        2020-02-02.
+        """
         for line_values in lines_data:
             date = self._get_utc_line_date(line_values)
             date = date.astimezone(utc).replace(tzinfo=None)
+            # Add amounts from dates before statement to balance_start.
             if date < statement_date_since:
                 if "balance_start" in statement_values:
                     statement_values["balance_start"] = Decimal(
                         statement_values["balance_start"]
                     ) + Decimal(line_values["amount"])
                 continue
-            elif date >= statement_date_until:
+            # Subtract amounts from dates after statement from balace_end_real.
+            if date >= statement_date_until:
```

Comment:
Why change this `elif` into an `if`? That would mean you expect that it could be that `statement_date_until < statement_date_since`, and that's something I would rather add an `assert` for that it does not happen.

Reply:
An `elif` combines an else with an if. The else part here is superfluous, but forces you to look at the other if to which the else is connected. So a pure if here does not cause any other funtionality, but makes the code more clear and readable.